java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSDeltalakeToIceberg.java

```diff
@@ -0,0 +1,147 @@
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GCSDeltalakeToIceberg.class);
+
+  private final GCSDeltalakeToIcebergConfig gcsDeltalakeToIcebergConfig;
+
+  public static GCSDeltalakeToIceberg of(String... args) {
+    GCSDeltalakeToIcebergConfig config =
+        GCSDeltalakeToIcebergConfig.fromProperties(PropertyUtil.getProperties());
+    LOGGER.info("Config loaded\n{}", config);
+    return new GCSDeltalakeToIceberg(config);
+  }
+
+  public GCSDeltalakeToIceberg(GCSDeltalakeToIcebergConfig config) {
+    this.gcsDeltalakeToIcebergConfig = config;
+  }
+
+  @Override
+  public void validateInput() throws IllegalArgumentException {
+    ValidationUtil.validateOrThrow(gcsDeltalakeToIcebergConfig);
+  }
+
+  @Override
+  public void runTemplate()
+      throws StreamingQueryException, TimeoutException, SQLException, InterruptedException {
+
+    LOGGER.info("Initialize Spark Session With Iceberg And Deltalake Properties");
+    SparkSession sparkSession =
+        SparkSession.builder()
+            .appName("GCSDeltalakeToIceberg Dataproc Job")
+            .config(
+                "spark.sql.extensions",
+                "io.delta.sql.DeltaSparkSessionExtension,org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
+            .config(
+                "spark.sql.catalog.spark_catalog", "org.apache.iceberg.spark.SparkSessionCatalog")
+            .config("spark.sql.catalog.spark_catalog.type", "hive")
+            .getOrCreate();
+
+    LOGGER.info("Set Log Level {}", gcsDeltalakeToIcebergConfig.getSparkLogLevel());
+    sparkSession.sparkContext().setLogLevel(gcsDeltalakeToIcebergConfig.getSparkLogLevel());
+
+    LOGGER.info("Read Deltalake Table From {}", gcsDeltalakeToIcebergConfig.getInputFileLocation());
+    int versionAsOf = Math.max(gcsDeltalakeToIcebergConfig.getVersionAsOf(), 0);
+    String timestampAsOf =
+        gcsDeltalakeToIcebergConfig.getTimestampAsOf() == null
+                || gcsDeltalakeToIcebergConfig.getTimestampAsOf().isEmpty()
+            ? null
+            : gcsDeltalakeToIcebergConfig.getTimestampAsOf();
+
+    Dataset<Row> dataset;
+    if (timestampAsOf != null) {
+
+      LOGGER.info(
+          "Time Travel By Timestamp Settings Detected With TimestampAsOf: {}", timestampAsOf);
+      dataset =
+          sparkSession
+              .read()
+              .format("delta")
+              .option("timestampAsOf", timestampAsOf)
+              .load(gcsDeltalakeToIcebergConfig.getInputFileLocation());
+    } else {
+
+      LOGGER.info("Time Travel By Version Settings Detected With VersionAsOf: {}", versionAsOf);
+      dataset =
+          sparkSession
+              .read()
+              .format("delta")
+              .option("versionAsOf", versionAsOf)
+              .load(gcsDeltalakeToIcebergConfig.getInputFileLocation());
+    }
+
+    LOGGER.info("Write Iceberg Table To {}", gcsDeltalakeToIcebergConfig.getIcebergTableName());
+    SaveMode saveMode =
```

Comment:
In all the other templates user is providing value of save mode and we are using it directly as string.
This feels like unnecessary if/else